### DISCUSSION TITLE
How to get notified when certain mutations succeed

### DISCUSSION THREAD

**Discussion by krachtstefan** [TanStack/query#6241] [General] [answered]
I try to keep my question as short as possible. Considering the circumstances in a web app:
- I have an endpoint that fetches a list of notifications (like the ones you have behind a notification bell)
- I have around 50 mutations in my app
- 10 of these mutations can cause some notifications to (dis)appear, so I need to refetch whenever they are successful

I would like to configure the refetches in one place (near that notification component) and avoid going to all 10 `onSuccess` callbacks and trigger a refetch. What could be a good strategy? Is there a way to have an `onSuccess` callback for certain `mutationKey`s.

Thanks in advance for your input!

**Comment by TkDodo** [comment, marked as answer]
you could use the new `useMutationState` hook in v5 to listen to all mutations, then find out when they have all finished and then fire an effect that invalidates.

otherwise, the global `onSuccess` callback of the `MutationCache` can also do it.

**Reply by krachtstefan** [reply]
I stumbled upon `useMutationState` in the V5 release notes and was wondering if this could be a solution. But I was just not sure. 

Thank's for clarifying @TkDodo and congrats to the v5 release, I will update very soon!

*after reading my question again, my first sentence could be read a bit rude. Please read this as "I appreciate anyone reading this and don't want to waste anyone's time, so I want to get straight to my question."
